lib/kits/core-ui/paged-publication.js

```diff
@@ -0,0 +1,206 @@
+        pageList: document.getElementById(
+            'sgn-sdk-paged-publication-viewer-page-list-template'
+        ),
+        menuContainer: document.getElementById(
+            'sgn-sdk-paged-publication-viewer-menu-container-template'
+        )
+    };
+
+    const renderMainContainer = () => {
+        const mainContainerEl = document.querySelector(scriptEls.mainContainer);
+
+        PagedPublicationComponents.MainContainer({
+            template: customTemplates.mainContainer,
+            shoppingListCounterTemplate: customTemplates.shoppingListCounter,
+            el: mainContainerEl,
+            scriptEls
+        }).render();
+    };
+    renderMainContainer();
+
+    const pagedPublicationObj = {
+        render: async () => {
+            if (Object.keys(options).length === 0) {
+                await pagedPublicationObj.setOptions();
+            }
+
+            pagedPublicationObj.start();
+
+            pagedPublicationObj.renderShoppingList();
+            pagedPublicationObj.renderMenuPopup();
+        },
+
+        renderShoppingList: () => {
+            PagedPublicationComponents.ShoppingList({
+                template: customTemplates.shoppingList
+            }).render();
+        },
+
+        renderMenuPopup: () => {
+            const menuBtn = document.querySelector('.sgn__nav-menu-btn');
+            menuBtn.addEventListener('click', (e) => {
+                e.preventDefault();
+                PagedPublicationComponents.MenuPopup({
+                    configs: options,
+                    sgnData,
+                    sgnViewer,
+                    templates: customTemplates
+                }).render();
+            });
+        },
+
+        animateShoppingListCounter: () => {
+            const shoppingListCounter = document.querySelector(
+                '.sgn__offer-shopping-list-counter'
+            );
+
+            shoppingListCounter?.classList.remove('sgn-animate-bounce');
+            shoppingListCounter?.classList.add('sgn-animate-bounce');
+        },
+
+        setOptions: async (opts = {}) => {
+            options = {
+                el: document.querySelector('.sgn__pp'),
+                eventTracker: SGN.config.get('eventTracker'),
+                pageId:
+                    opts.pageId ||
+                    `page${SGN.util.getQueryParam(scriptEls.pageIdParam) || 1}`,
+                id:
+                    opts.id ||
+                    SGN.util.getQueryParam(scriptEls.publicationIdParam) ||
+                    scriptEls.publicationId ||
+                    '',
+                businessId: opts.businessId || scriptEls.businessId
+            };
+
+            if (!options.id) {
+                options.id =
+                    await pagedPublicationObj.fetchLatestPublicationId();
+            }
+        },
+
+        start: () => {
+            const bootstrapper = new PagedPublicationKit.Bootstrapper(options);
+
+            bootstrapper.fetch((err, data) => {
```

Comment:
You could skip `bootstrapper.fetch` here since you already can have the relevant publication in hand with `.fetchLatestPublication`. Just fetch `catalogs/${publication.id}/pages` too and then you can call into `createViewer` like this:

```js
sgnViewer = bootstrapper.createViewer({
    details: publication,
    pages: await request({url: `catalogs/${publication.id}/pages`})
});
```

Saves an unnecessary `catalogs/${publication.id}` request that `bootstrapper.fetch` wastes its time doing here

Reply:
Do you think it will be fine if we maintain this as it is for now? `.fetchLatestPublication` is only a fallback when there's no available publicationId being passed as parameter (url param, or param from publications list view).